src/data/i18n.test.tsx

```diff
@@ -42,3 +43,73 @@ test('getTranslations gets a list of language versions', test => {
     { locale: DE, word: 'hund' },
   ])
 })
+
+test('findTranslationRecursively on objects', test => {
+  const heroSection = {
+    title: 'heroSection',
+    translations: [
+      { locale: EN, header: 'Welcome' },
+      { locale: DE, header: 'Wilkommen' },
+    ],
+  }
+
+  const landingPage = {
+    title: 'landingPage',
+    translations: [
+      {
+        locale: EN,
+        heroSection: heroSection,
+      },
+      {
+        locale: DE,
+        heroSection: heroSection,
```

Comment:
If I'm not mistaken, the input data can only look like this if the user has typed the same heroSection in twice: once inside the german version of the landing page, and once inside the english.

Reply:
That's not a criticism of the function in any way

Reply:
True, the amount of redundancy.

Maybe it makes sense to restructure, e.g. : 
```
{
  translations: [
    {
      locale: [DE, EN],
      heroSection: heroSection
    }
  ]
}
```

Or just ignore the translation on top level and just defined a pure container class as such:
```
{
  title: "homepage"
  components: [
    heroSection: heroSection,
    offerings: offerings,
    ...
  ]
}
```